### PR TITLE
highlight DNS zone (BIND) files

### PR DESCRIPTION
**Pull request by mcandre**
Use Bind.tmbundle to highlight DNS zone (BIND) files, per #2364

### CODE REVIEW

**Comment by pchaigno** [comment]
Unfortunately, that won't be sufficient on its own. We'll have to define DNS zone as a language in Linguist. Don't worry, that's pretty easy. You just have to follow [these guidelines](https://github.com/github/linguist/blob/master/CONTRIBUTING.md#adding-a-language).

We don't already have a language using the `.zone` extension so you don't have to worry about that last part ;)

**Comment by mcandre** [comment]
@pchaigno Better?

File: lib/linguist/languages.yml [hunk 1]
```diff
@@ -3633,3 +3633,8 @@ xBase:
   - .prg
   tm_scope: none
   ace_mode: text
+
+zone:
```
Comment (pchaigno) [inline lib/linguist/languages.yml]: `DNS zone` would probably be better here ;)

**Comment by pchaigno** [comment]
> @pchaigno Better?

Much but there's still a few changes to do :smile: 
You'll have to specify the `tm_scope` and `ace_mode` fields for your `DNS zone` entry in `languages.yml` (`xBase` above for an example). `tm_scope` corresponds to the scope from your grammar (probably `text.zone`). You can leave `ace_mode` to `text`.

File: lib/linguist/languages.yml [hunk 2]
```diff
@@ -769,6 +769,13 @@ Diff:
   tm_scope: source.diff
   ace_mode: diff
 
+DNS zone:
+  type: programming
```
Comment (pchaigno) [inline lib/linguist/languages.yml]: We'll define it as a `data` language since these are basically data files ;)

**Comment by mcandre** [comment]
@pchaigno Cheers, I couldn't find "configuration", so I mistakenly chose "programming" at first.

**Comment by pchaigno** [comment]
Could you add the grammar using the `https` link? Travis doesn't seem to like SSH :/

**Comment by arfon** [comment]
Hi @mcandre - thanks for your work on this Pull Request. It looks like there are a few tests failing that we'll need to get fixed up before this can be merged:

```
1) Failure:
TestPedantic#test_language_names_are_sorted [/home/travis/build/github/linguist/test/test_pedantic.rb:9]:
Diff should come after DNS zone
```

Please could you fix the ordering position of `DNS zone`?

```
2) Failure:
TestGrammars#test_submodules_have_licenses [/home/travis/build/github/linguist/test/test_grammars.rb:106]:
The following grammar submodules don't seem to have a license. All grammars must have a license that permits redistribution.
vendor/grammars/Bind.tmbundle
```

It looks like the grammar repository doesn't have a license which means we can't include it in Linguist. Would you mind opening an issue on the grammar repository to ask them to include a license?

```
3) Failure:
TestBlob#test_language [/home/travis/build/github/linguist/test/test_blob.rb:519]:
sneaky.net.zone.
Expected: "zone"
  Actual: "DNS zone"
```

It looks like the sample file `sneaky.net.zone` needs moving to the correct samples directory.

screen shot 2015-05-21 at 11 07 07 am

**Comment by bkeepers** [comment]
:wave: @mcandre looks like there still a failing test. Want to take a look at fixing that?

**Comment by larsbrinkhoff** [comment]
`.arpa` and `.db` are also used as file extensions for zone files.

**Comment by arfon** [comment]
@mcandre - are there any other bundles we could use for highlighting here? (given the lack of response from the author in https://github.com/norm/Bind.tmbundle/issues/1)

**Comment by mcandre** [comment]
@arfon Sadly, I don't know of any other bundles.

**Comment by arfon** [comment]
@mcandre what about this one? https://github.com/sixty4k/st2-zonefile - might be worth asking them to add a license to this repo?

**Comment by sixty4k** [comment]
Added MIT License to st2-zonefile.

**Comment by arfon** [comment]
> Added MIT License to st2-zonefile. 

Awesome. Thanks @sixty4k!

@mcandre - would you be able to update this PR using @sixty4k's bundle?

**Comment by arfon** [comment]
[Prettylights ](https://github-lightshow.herokuapp.com/?utf8=%E2%9C%93&scope=from-url&grammar_url=https%3A%2F%2Fgithub.com%2Fsixty4k%2Fst2-zonefile%2Fblob%2Fmaster%2FBind%2520Zone%2520Files.tmLanguage&grammar_text=&code_source=from-url&code_url=https%3A%2F%2Fgithub.com%2Fmcandre%2Fdocker-bind%2Fblob%2Fmaster%2Fsneaky.net.zone&code=) looks pretty reasonable.

**Comment by arfon** [comment]
Closing in favour of https://github.com/github/linguist/pull/2554

**Pull request closed without merging**